### PR TITLE
separate out gn fake platform declaration

### PR DESCRIPTION
**Pull request by andy31415**
#### Problem
`./gn_build.sh` does not run anymore due to fake platform dependencies

#### Change overview

Create separate fake toolchain and only include that in the top level build.gn (example builds will not pick it up as a viable platform unless they include top level or explicitly add the platform.

#### Testing

Ran `./gn_build.sh` on linux
build rpc_console on darwin using build_examples.py

### CODE REVIEW

**Comment by andy31415** [comment]
Changed to draft: darwin rpc_console compilation compiles after this change (cannot find device.gni)

**Comment by andy31415** [comment]
Fast track: hotfix really, tested with builds on linux and darwin.

**Review by cecille — APPROVED** [review]

File: build/toolchain/host/BUILD.gn [hunk 1]
```diff
@@ -33,14 +34,12 @@ gcc_toolchain("${host_os}_${host_cpu}_clang") {
...
-if (defined(chip_fake_plaform)) {
-  gcc_toolchain("fake_${host_cpu}_gcc") {
-    toolchain_args = {
-      current_os = host_os
-      current_os = host_os
-      current_cpu = host_cpu
-      is_clang = false
-      chip_fake_platform = true
-    }
+gcc_toolchain("fake_${host_cpu}_gcc") {
```
Comment (mspang) [inline build/toolchain/host/BUILD.gn:37]: Fix it by moving this to a separate build file.

**Review by mspang — CHANGES_REQUESTED** [review]

File: build/toolchain/host/BUILD.gn (hunk 1 above)
Comment (andy31415) [reply build/toolchain/host/BUILD.gn]: Done

**Review by mspang — APPROVED** [review]

**Pull request merged**